src/library-authoring/create-container/CreateContainerModal.test.tsx

```diff
@@ -0,0 +1,241 @@
+import React from 'react';
+import { initializeMockApp } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import MockAdapter from 'axios-mock-adapter';
+
+import { useLibraryContext } from '../common/context/LibraryContext';
+import { ToastContext, ToastContextData } from '../../generic/toast-context';
+import CreateContainerModal from './CreateContainerModal';
+import { ContainerType } from '../../generic/key-utils';
+import { getLibraryCollectionItemsApiUrl, getLibraryContainerChildrenApiUrl } from '../data/api';
+
+let axiosMock: MockAdapter;
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+    },
+  },
+});
+
+const mockNavigateTo = jest.fn();
+const mockShowToast = jest.fn();
+const mockCloseToast = jest.fn();
+
+// Mock the useLibraryRoutes hook
+const mockUseLibraryRoutes = jest.fn().mockReturnValue({
+  navigateTo: mockNavigateTo,
+  insideCollection: true,
+  insideSection: false,
+  insideSubsection: false,
+});
+
+jest.mock('../routes', () => ({
+  useLibraryRoutes: () => mockUseLibraryRoutes(),
+}));
+
+// Mock the useLibraryContext hook
+jest.mock('../common/context/LibraryContext', () => ({
+  useLibraryContext: jest.fn(),
+}));
+
+const defaultLibraryContext = {
+  libraryId: 'test-library',
+  collectionId: 'test-collection',
+  containerId: 'lct:org:lib:unit:test-container',
+  setContainerId: jest.fn(),
+  readOnly: false,
+  isLoadingLibraryData: false,
+  showOnlyPublished: false,
+  extraFilter: [],
+  isCreateCollectionModalOpen: false,
+  openCreateCollectionModal: jest.fn(),
+  closeCreateCollectionModal: jest.fn(),
+  createContainerModalType: ContainerType.Chapter,
+  setCreateContainerModalType: jest.fn(),
+  componentBeingEdited: undefined,
+  openComponentEditor: jest.fn(),
+  closeComponentEditor: jest.fn(),
+  setCollectionId: jest.fn(),
+};
+
+const defaultToastContext: ToastContextData = {
+  showToast: mockShowToast,
+  closeToast: mockCloseToast,
+  toastMessage: null,
+  toastAction: undefined,
+};
+
+const renderComponent = (
+  toastContextValue = defaultToastContext,
+) => {
+  // Set up the mock implementation for useLibraryContext
+  jest.mocked(useLibraryContext).mockReturnValue(defaultLibraryContext);
```

Comment:
I think it's better to use something [like this](https://github.com/openedx/frontend-app-authoring/blob/d599473f5a99e52f70532da83f39bbbaf84a6c8d/src/library-authoring/LibraryContent.test.tsx#L50)